tests/regtest.py

```diff
@@ -1149,6 +1183,56 @@ def run(self):
 
 			print()
 
+class TestCorrectOwner(TestBase):
+
+	def __init__(self):
+		super().__init__("TestCorrectOwner", "checks whether file owner/group assignments are corectly applied as configured")
+
+	def run(self):
+
+		if self.complainOnNoSubIdSupport():
+			# we need sub-uids to test ownership changes
+			return
+
+		testdir = self.createAndGetTestDir(0o770)
+
+		# don't use friendly user and group names but plain numerical
+		# IDs instead. This way we don't have to adjust /etc/passwd
+		# and /etc/group. Numerical IDs are only supported in newer
+		# chkstat versions.
+
+		# we need a defined order of execution here, therefore
+		# iterate over the sorted dictionary keys.
+		#
+		# we start out from 0:0 and downgrade first to 0:1 then to 1:1
+		# to avoid triggering the "refusing to correct" logic in
+		# chkstat.
+		owners = {
+			"easy": ("0", "0"),
+			"paranoid": ("0", "1"),
+			"secure": ("1", "1"),
+		}
+
+		entries = {}
+
+		for profile in sorted(owners.keys()):
+			user, group = owners[profile]
+			lines = entries.setdefault(profile, [])
```

Comment:
Yeah `setdefault` is unnecessary. It's a slip since I copied over template code from another test. I'll keep the loop though, don't want to make the code too compact, also hurts readability sometimes.